pkg/api/ref.go

```diff
@@ -52,23 +51,41 @@ func GetReference(obj runtime.Object) (*ObjectReference, error) {
 	if err != nil {
 		return nil, err
 	}
-	_, kind, err := Scheme.ObjectVersionAndKind(obj)
-	if err != nil {
-		return nil, err
+
+	// if the object referenced is actually persisted, we can just get kind from meta
+	// if we are building an object reference to something not yet persisted, we should fallback to scheme
+	kind := meta.Kind()
+	if kind == "" {
+		_, kind, err = Scheme.ObjectVersionAndKind(obj)
```

Comment:
Add TODO to the method to require a meta.Interface() and spawn an issue and I'll allow it.